core/src/main/java/org/mule/runtime/core/api/exception/WildcardErrorTypeMatcher.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.api.exception;
+
+import java.util.Objects;
+
+import org.mule.runtime.api.message.ErrorType;
+
+public final class WildcardErrorTypeMatcher implements ErrorTypeMatcher {
+
+  public static String WILDCARD_TOKEN = "*";
+
+  private final ErrorType errorType;
+
+  private boolean identifierIsWildcard;
+
+  private boolean namespaceIsWildcard;
+
+  public WildcardErrorTypeMatcher(ErrorType errorType) {
+    this.errorType = errorType;
+    identifierIsWildcard = WILDCARD_TOKEN.equals(errorType.getIdentifier());
+    namespaceIsWildcard = WILDCARD_TOKEN.equals(errorType.getNamespace());
+  }
+
+  @Override
+  public boolean match(ErrorType errorType) {
+    if (matchIdentifier(errorType) && matchNamespace(errorType) && matchParent(errorType)) {
+      return true;
+    }
+
+    return isChild(errorType);
+  }
+
+  private boolean matchParent(ErrorType errorType) {
```

Comment:
How do you know who the parent is from something like "HTTP:*" or "*:CONNECTIVITY"?